it/src/main/java/com/google/cloud/teleport/it/jdbc/AbstractJDBCResourceManager.java

```diff
@@ -0,0 +1,363 @@
+   * that id. i.e. {0: [val1, val2, ...], 1: [val1, val2, ...], ...}
+   *
+   * @param tableName The name of the table to insert the given rows into.
+   * @param rows A map representing the rows to be inserted into the table.
+   * @param columns A list of columns to write the mapped values to.
+   * @throws JDBCResourceManagerException if method is called after resources have been cleaned up,
+   *     if the manager object has no dataset, if the table does not exist or if there is an
+   *     Exception when attempting to insert the rows.
+   */
+  public boolean write(String tableName, Map<Integer, List<Object>> rows, List<String> columns) {
+    LOG.info("Attempting to write {} rows to {}.{}.", rows.size(), databaseName, tableName);
+
+    if (rows.size() == 0) {
+      return false;
+    }
+
+    StringBuilder sql = new StringBuilder();
+    try (Connection con = driver.getConnection(getUri(), username, password)) {
+      Statement stmt = con.createStatement();
+      sql.append("INSERT INTO ").append(tableName);
+      if (columns.size() > 0) {
+        sql.append("(").append(String.join(",", columns)).append(")");
+      }
+      sql.append(" VALUES ");
+
+      List<String> rowsToInsert = new ArrayList<>();
+      for (Integer rowId : rows.keySet()) {
+        StringBuilder valueList = new StringBuilder("(");
+
+        List<String> rowToInsert = new ArrayList<>();
+        rowToInsert.add(String.valueOf(rowId));
+        for (Object value : rows.get(rowId)) {
+          if (!NumberUtils.isCreatable(value.toString())) {
+            rowToInsert.add("'" + value + "'");
+          } else {
+            rowToInsert.add(String.valueOf(value));
+          }
+        }
+        valueList.append(String.join(",", rowToInsert)).append(")");
+
+        rowsToInsert.add(valueList.toString());
+      }
+      stmt.executeUpdate(sql.append(String.join(",", rowsToInsert)).toString());
+
+    } catch (Exception e) {
+      throw new JDBCResourceManagerException(
+          "Failed to insert values into table with SQL statement: " + sql, e);
+    }
+
+    LOG.info("Successfully wrote {} rows to {}.{}.", rows.size(), databaseName, tableName);
+
+    return true;
+  }
+
+  @Override
+  public Map<Integer, List<String>> readTable(String tableName) {
+    LOG.info("Reading all rows from {}.{}", databaseName, tableName);
+
+    Map<Integer, List<String>> resultMap = new HashMap<>();
+
+    StringBuilder sql = new StringBuilder();
+    try (Connection con = driver.getConnection(getUri(), username, password)) {
+      Statement stmt = con.createStatement();
+
+      sql.append("SELECT * FROM ").append(tableName);
+      ResultSet result = stmt.executeQuery(sql.toString());
+
+      while (result.next()) {
+        int id = result.getInt(tableIds.get(tableName));
+        List<String> rowToInsert = new ArrayList<>();
+        for (int i = 2; i <= result.getMetaData().getColumnCount(); i++) {
+          rowToInsert.add(result.getString(i));
+        }
```

Comment:
Can we handle this better than by the index? i.e., scanning all columns and treating `tableIds.get(tableName)` differently

Reply:
Done.